### ISSUE TITLE
Particle filter trajectories all the same 

### ISSUE THREAD

**Issue by adamkucharski** [LloydChapman/seir#1]
In `fit_seirdage.Rmd`, the particle filter seems to generate identical trajectories for all particles in [this line of code](https://github.com/LloydChapman/seir/blob/7f13f3ae4ed47eb2896277a6d9fd50bf9c5af911/vignettes/fit_seirdage.Rmd#L139).

This lack of stochastic variation between particle trajectories is also apparent in the pMCMC, which seems to only capture parameter uncertainty, rather than the reconstructed trajectory from the filter:
<img width="674" alt="Screenshot" src="https://github.com/LloydChapman/seir/assets/8329046/f8cddc40-d88a-4070-963a-143d4b95a99c">
I wondered if it might be something to do with `seed=1` defined in [the earlier model](https://github.com/LloydChapman/seir/blob/7f13f3ae4ed47eb2896277a6d9fd50bf9c5af911/vignettes/fit_seirdage.Rmd#L60), but the later filter and pMCMC uses underlying `seirdage`, so this shouldn't be the problem?

The [original odin model](https://github.com/LloydChapman/seir/blob/7f13f3ae4ed47eb2896277a6d9fd50bf9c5af911/inst/odin/seirhdage.R#L54) has random draws between transitions, so not clear to me why the simulations aren't different – but I'm probably missing something obvious!

**Comment by LloydChapman** [comment]
I've looked into this quite thoroughly and it's a particle degeneracy problem. The particle filter actually generates slightly different trajectories, but the differences are very small and only really in the values at the last few time points. 

This is due to simulating data for a large population O(10^6) and assuming the observed incidence is Poisson distributed with mean given by the modelled incidence, which corresponds to a very narrow likelihood function around the modelled incidence. As the likelihood function is so narrow, at each observation nearly all the particles get very low weights and are lost in the resampling, such that all the particles essentially end up being the same.

I've fixed this by changing the likelihood function to be negative binomial, so that the particle MCMC now at least works, albeit still with poor mixing. I've also changed the observed simulated data to be negative binomially distributed, for consistency, and changed the vignette to use a MCMC run of the "deterministic" version of the model (with random draws replaced by expectations) to tune the MCMC, as this is the approach Imperial have used for more complicated models in their [papers](https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-023-39661-5/MediaObjects/41467_2023_39661_MOESM1_ESM.pdf).
